### PR TITLE
[RFC] Jenkinsfile: reduce duplicated code by defining some functions

### PR DESCRIPTION
**Pull request by thaJeztah**
not sure if this works, but an attempt at reducing a lot of duplication in our Jenkinsfile

### CODE REVIEW

**Comment by thaJeztah** [comment]
ok, first attempt didn't work

```
WorkflowScript: 145: Method calls on objects not allowed outside "script" blocks. @ line 145, column 33.
                                   utils.buildDevImage()
                                   ^
```

**Comment by thaJeztah** [comment]
some examples here; https://stackoverflow.com/questions/47628248/how-to-create-methods-in-jenkins-declarative-pipeline/47631522
and here https://stackoverflow.com/questions/47628248/how-to-create-methods-in-jenkins-declarative-pipeline/47631522

File: Jenkinsfile [hunk 1]
```diff
@@ -1,4 +1,45 @@
 #!groovy
+
+void printInfo() {
```
Comment (thaJeztah) [inline Jenkinsfile]: I tried to move this to a separate file (which might be possible, but let's discuss first if we like the approach)

**Comment by thaJeztah** [comment]
@andrewhsu @tiborvass wdyt of an approach like this?

**Comment by tiborvass** [comment]
@thaJeztah I think that I would not oppose this, but I would much rather go away from Jenkinsfile as much as possible, so that we can simply call make targets instead, the same ones that would be called in CI from Jenkinsfile.

**Comment by thaJeztah** [comment]
@tiborvass agreed; we should try to move things to the makefile as much as possible. Not sure how we'll deal with Windows (without having `make` there; I think there's some "make-like" tools though).

There might still be CI-specific steps (not sure if we want those in the makefile)

I can close this one for now, as it was mainly a PoC / quick attempt

**Comment by tiborvass** [comment]
@thaJeztah we could either leverage bash on windows, or start converting bash to Go.

**Comment by thaJeztah** [comment]
(not related to testing, but also thinking how important it is to support build everything outside of a container, or if we can start inlining things in the dockerfile)

**Pull request closed without merging**